frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -277,24 +278,51 @@ class NativeQueryEditor extends Component {
             Date.now() - timestamp < AUTOCOMPLETE_CACHE_DURATION &&
             this._lastAutoComplete.prefix === prefix;
           if (!cacheHit) {
+            // Get models and fields from tables
             // HACK: call this.props.autocompleteResultsFn rather than caching the prop since it might change
-            results = await this.props.autocompleteResultsFn(prefix);
+            const apiResults = await this.props.autocompleteResultsFn(prefix);
             this._lastAutoComplete = {
               timestamp: Date.now(),
               prefix,
               results,
             };
+
+            // Get referenced questions
+            const referencedQuestionIds =
+              this.props.query.referencedQuestionIds();
+            // The results of the API call are cached by ID
+            const referencedQuestions = await Promise.all(
+              referencedQuestionIds.map(id => this.props.fetchQuestion(id)),
+            );
+
+            // Get columns from referenced questions that match the prefix
+            const lowerCasePrefix = prefix.toLowerCase();
+            const isMatchForPrefix = name =>
+              name.toLowerCase().includes(lowerCasePrefix);
+            const questionColumns = referencedQuestions
+              .filter(Boolean)
+              .flatMap(question =>
+                question.result_metadata
+                  .filter(columnMetadata =>
+                    isMatchForPrefix(columnMetadata.name),
+                  )
+                  .map(columnMetadata => [
+                    columnMetadata.name,
+                    `${question.name} :${columnMetadata.base_type}`,
+                  ]),
+              );
+
+            // Concat the results from tables, models, and referenced questions
+            results = apiResults.concat(questionColumns);
```

Comment:
I noticed that it seems to remove dup'd options when the `name` is the same, so if a model has fields with names that are identical to table names, they won't show.

Perhaps we could swap this concat around to `questionColumns.concat(apiResults)` to try to ensure that they when the race

Reply:
Great point!